a/render-test/main.lisp

```diff
@@ -124,39 +150,52 @@
-               (s:compile-shader shader)
-               (when (eq (s:get-status shader)
-                         :error)
-                 (log:info "Shader compilation failed")
-                 (log:info (s:get-compile-log shader)))
-               (when (eq (s:get-status shader)
-                         :compiled)
-                 (log:info "Shader successfully compiled and linked!"))
-               (let ((log (s:get-compile-log shader)))
-                 (when (> (length log) 0)
-                   (log:info "Compilation log: ~A" log)))
-               (b:use-shader (r:renderer-backend r) shader)
-               ;; Texture
-               (t:texture-load texture)
-               (setf (slot-value (r:renderer-backend r) 'lbge.render.gl::active-texture)
-                     texture)
                (gl:polygon-mode :front-and-back :fill))) ; change to :line to view wireframe
     (timer:link timer (lambda ()
+                        (log:debug "FPS ~A (~A frames ~A seconds)"
+                                   (/ (b:get-total-frames (r:renderer-backend r))
+                                      (/ total-dt 1000.0))
+                                   (b:get-total-frames (r:renderer-backend r))
+                                   (/ total-dt 1000.0))
                         (b:print-statistics (r:renderer-backend r))))
     (le:link :on-loop
-             (lambda (dt)
-               (let ((backend (r:renderer-backend r)))
-                 (an:update animation dt)
-                 (b:clear backend)
-                 (b:render backend r)
-                 (b:present backend)
-                 (timer:update timer dt)))))
+             (let ((dr 0.02)
+                   (dg 0.01)
+                   (db 0.03))
+               (lambda (dt)
+                 (let ((backend (r:renderer-backend r)))
+                   (let ((color (color plain)))
+                     (incf (m:x color) dr)
+                     (incf (m:y color) dg)
+                     (incf (m:z color) db)
+                     (when (> (m:x color) 1.0)
+                       (setf
+                        dr (- dr)
+                        (m:x color) (+ 1.0 dr)))
+                     (when (> (m:y color) 1.0)
+                       (setf
+                        dg (- dg)
+                        (m:y color) (+ 1.0 dg)))
+                     (when (> (m:z color) 1.0)
+                       (setf
+                        db (- db)
+                        (m:z color) (+ 1.0 db)))
+                     (when (< (m:x color) 0.0)
+                       (setf
+                        dr (- dr)
+                        (m:x color) (+ 0.0 dr)))
+                     (when (< (m:y color) 0.0)
+                       (setf
+                        dg (- dg)
+                        (m:y color) (+ 0.0 dg)))
+                     (when (< (m:z color) 0.0)
+                       (setf
+                        db (- db)
+                        (m:z color) (+ 0.0 db))))
```

Comment:
This should really be `macro`ized

Reply:
This will most probably be replaced with proper bounded animation.
I have a couple of ideas.